pkg/util/utils.go

```diff
@@ -142,7 +142,12 @@ func GVRFromType(resourceName string, expectedType interface{}) *schema.GroupVer
 		// testUnstructuredMock.Foo is a mock type for testing
 		return nil
 	}
-	apiVersion := expectedType.(*unstructured.Unstructured).Object["apiVersion"].(string)
+	t, err := meta.TypeAccessor(expectedType)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get type accessor", "expectedType", expectedType)
+		return nil
```

Comment:
Apologies to block this again, but since this is used in a bunch of critical paths in CRS procedures, continuing silently after an error wouldn't make sense. I'd push for changing the signature here, and surfacing this up, as pointed in my previous review.